lib/components/base/Badge.vue

```diff
@@ -53,16 +53,15 @@ import {
   capitalizeString,
 } from '@'
 
-defineProps({
-  type: {
-    type: String,
-    required: true,
-  },
-  color: {
-    type: String,
-    default: '',
-  },
-})
+withDefaults(
+  defineProps<{
+    type: string
+    color: string
+  }>(),
+  {
+    color: '',
```

Comment:
No need for an empty default if you change the class binding on line 2 from
```vue
<span :class="'version-badge ' + color + ' type--' + type">
```
to
```vue
<span :class="['version-badge', color, `type--${type}`]">
```

It will also be in conflict with narrowed union type for `color` per suggestion above.